aicssegmentation/structure_wrapper/WorkflowEngine.py

```diff
@@ -0,0 +1,139 @@
+from aicssegmentation.structure_wrapper_config.structure_config_utils import load_workflow_config, parse_config_to_objects
+import numpy as np
+from aicssegmentation.structure_wrapper.WorkflowStep import WorkflowStep
+
+
+class WorkflowEngine:
+    """
+    A class to define a whole aics-segmentation workflow
+    """
+    def __init__(self, workflow_name: str, image: np.ndarray):
+        """
+        Constructor for the WorkflowEngine object
+
+        Params:
+            workflow_name (str): dictionary object containing information about this workflow step
+            image (np.ndarray):  image to perform workflow on
+        """
+        self.workflow_name: str = workflow_name   # Workflow name
+        self.steps: list = self.__get_steps()        # List of WorkflowSteps for this workflow
+        self.currentStep: int = 0                 # Next step to execute
+        self.starting_image: np.ndarray = image          # Initial image
+
+    def __get_steps(self) -> list:
+        """
+        Get a list of WorkflowStep objects to perform on the starting image.
+
+        Params:
+            none
+
+        Returns:
+            (list(WorkflowStep)): List of workflow step objects
+       """
+        # TODO: in order for parent fucntionality to work correctly, we should sort these in the list by parent index
+        return parse_config_to_objects(load_workflow_config(self.workflow_name))
+
+    def get_next_step(self) -> WorkflowStep:
+        """
+        Get the next step to be performed
+
+        Params:
+            none
+
+        Returns:
+            (WorkflowStep): next WorkflowStep object to perform on image
+       """
+        return self.steps[self.currentStep]
+
+    def execute_next(self) -> np.ndarray:
+        """
+        Execute the next workflow step.
+
+        Params:
+            none
+
+        Returns:
+            result (np.ndarray): resultant image from running the next workflow step
+       """
+        print("running step")
+        print(self.currentStep)
+        # Pick which image to perform the workflow step on
+        image: np.ndarray = None
+        if self.currentStep == 0:
+            # First image, so use the starting image for the next workflow step
+            image = self.starting_image
+        elif self.isDone():
+            # No more workflow steps to perform
+            # TODO: what to do if done with workflow but execute_next is prompted?
+            # printing message for now
+            print("No steps left to run")
+        else:
+            # First step has been run, so run next workflow step with the result of its parent
+            image = self.get_result(self.get_next_step().parent)
```

Comment:
I don't know if this is something that should be solved in the engine or in how the segmentation functions are implemented, but right now we have some steps that return multiple images [(e.g Masked Object Thresholding)](https://github.com/AllenCell/aics-segmentation/blob/feature/json_workflow/aicssegmentation/core/MO_threshold.py) and subsequent steps that only rely on one of those images [(e.g.  NPM1)](https://github.com/AllenCell/aics-segmentation/blob/feature/json_workflow/aicssegmentation/structure_wrapper_config/conf_npm1.json) @jxchen01 should I split the MO threshold into two functions to avoid this problem?

Reply:
i will take a look. Thanks for bringing this up

Reply:
@benjijamorris The second output of MO is only for debugging. So, i think at least for MO, we can return a single output. Are you aware of any other function that returns multiple outputs?

Reply:
I think that's the only one we're using now